### ISSUE TITLE
Planning for the possibility of WeScheme Deprecation

### ISSUE THREAD

**Issue by flannery-denny** [bootstrapworld/curriculum#1271]
I woke up thinking about @schanzer's concern around Google breaking WeScheme this spring and how we could support teachers like myself of the past who are dedicated to Bootstrap from way back and unfamiliar with Pyret... and I think it's pretty straight forward.  I would suggest that preparing for this eventuality in advance wouldn't be a huge lift and will make a big difference in retaining teachers and decreasing stress for our team members.

**What I would definitely need to keep teaching:**

- [ ] A one page cheat sheet translating what I know from WeScheme into Pyret 
- operators - with acknowledgement that they're treated as functions in Scheme
- functions
- how to teach circles of evaluation - it was really hard for me to make sense of how to use the ant I was so comfortable teaching with in scheme in pyret
- the exciting acknowledgement that I would now have access to Data Science lessons
- [ ] A link to the contracts pages (we already have these)
- [ ] A fully functioning sample game code, commented as if I'd taken good notes in class (because my coding was always rusty when we first started the unit and the game code I made in my training was my reference)
- [ ] Encouragement to work through the game starter file and make a new game code. 
- [ ] Solution pages (we already have these)
- [ ] **Warning!**
- An email linking to a page on our website with the above resources, giving me a heads up that I'm going to need to carve out a few hours to prepare, before the crisis moment!  
-- I'd suggest multiple emails: one this spring, one early summer and one at the begin of the school year, because not everyone opens all their emails and this would be BIG NEWS that we want to make sure people don't miss.
- A message when I tried to open wescheme.org 
- A banner message when I tried to open archived versions of Bootstrap materials
- [ ] free or cheap access to a printed workbook (perhaps even a bonus Data Science workbook to tantalize me)
- any chance there's funding for this? I remember having gotten an email about a donor underwriting free workbooks at some point in the past and engaging with it.

**What else might support me?**
- [ ] An invitation to a half-day virtual pd to support me in transferring what I know (and hyping me on the open door to Data Science materials, which I might then sign up to train on)
- Strongly suggest making these free or cheap enough that teachers could pay it out of pocket without funding from their schools _(even if we only offered a free workshop once, and I couldn't make it, I would know that Bootstrap had really had my back!)_  
- perhaps fees could cover shipping teachers copies of both our algebra and data science workbooks?
- [ ] An invitation to join a special discourse space for supporting WeScheme teachers transitioning to Pyret (which could be pre-populated with some posts hyping some things I probably missed like... 
- Computing Needs All Voices
- Hour of Code - parlay
- Hour of Data 
- How to sign up for a Data Science training for teachers already fluent in Bootstrap:Algebra in WeScheme

**If we do this thoughtfully, I honestly think it's a huge opportunity to engage our longtime teachers...**
Someone like me would be aware that there were new materials I hadn't carved out time to engage with, and losing WeScheme access would be the motivation to make time to use them, get me on discourse, and hyped about building my own pathway to incorporate some of our DS materials.

**Comment by retabak** [comment]
Wow, some really exciting ideas here! 
Loving all of this.

**Comment by schanzer** [comment]
The new WeScheme is running in test mode over at https://luke-dot-wescheme-hrd-2.appspot.com/. We found two regressions in testing, both of which are now fixed. Luke is also working on an improvement to the loading of very-long-program-lists, which takes nearly a full minute for my account.

At this point I am confident that we've got a working replacement. We'll switch over to the new system this Friday, giving us the weekend for any other bugs to shake out. But at this point we are definitely _not_ going to deprecate WeScheme, so this issue can be closed.
